renderer/src/features/mcp-servers/components/remote-mcp/__tests__/dialog-form-remote-mcp.test.tsx

```diff
@@ -265,6 +265,95 @@ describe('DialogFormRemoteMcp', () => {
     })
   })
 
+  it.each([
+    ['Dynamic Client Registration', 'none'],
+    ['OIDC', 'oidc'],
+  ])(
+    'submits Issuer URL when auth type is %s',
+    async (authOptionLabel, expectedAuthType) => {
+      const user = userEvent.setup({ delay: null })
+      const mockInstallServerMutation = vi.fn()
+
+      mockUseRunRemoteServer.mockReturnValue({
+        installServerMutation: mockInstallServerMutation,
+        isErrorSecrets: false,
+        isPendingSecrets: false,
+      })
+
+      renderWithProviders(
+        <Wrapper>
+          <DialogFormRemoteMcp
+            isOpen
+            closeDialog={vi.fn()}
+            groupName="default"
+          />
+        </Wrapper>
+      )
+
+      await waitFor(() => {
+        expect(screen.getByRole('dialog')).toBeVisible()
+      })
+
+      await user.type(
+        screen.getByRole('textbox', {
+          name: /server name/i,
+        }),
+        'issuer-enabled-server'
+      )
+
+      await user.type(
+        screen.getByRole('textbox', {
+          name: /server url/i,
+        }),
+        'https://api.example.com/mcp'
+      )
+
+      await user.type(screen.getByLabelText('Callback port'), '7777')
+
+      await user.click(screen.getByLabelText('Authorization method'))
+      await user.click(
+        screen.getByRole('option', {
+          name: authOptionLabel,
+        })
+      )
+
+      const issuerValue =
+        expectedAuthType === 'none'
+          ? 'https://issuer.example.com/none'
+          : 'https://issuer.example.com/oidc'
+
+      const issuerInput = screen.getByPlaceholderText(
+        'e.g. https://auth.example.com/'
+      )
+      await user.clear(issuerInput)
+      await user.type(issuerInput, issuerValue)
+
+      if (expectedAuthType === 'oidc') {
+        const clientIdInput = screen.getByPlaceholderText(
+          'e.g. 00000000-0000-0000-0000-000000000000'
+        )
+        await user.clear(clientIdInput)
+        await user.type(clientIdInput, 'oidc-client-id')
+      }
+
+      await user.click(screen.getByRole('button', { name: 'Install server' }))
+
+      await waitFor(() => {
```

Comment:
i guess that we could use `recordRequests` in order to make an assertion about the "data" attribute on the actual request payload instead of mocking the mutation. that way probably the assertion can also be simplified (fewer "objectContaining" and such)

but it's more of a nitpick than anythign

Reply:
in this case it is good to list complaints about it, because the current solution is extremely minimalistic and we can definitely improve it.